### PR TITLE
Tirtha/sticky notes

### PR DESCRIPTION
**Pull request by Tirtha1388**
Creating new PR for Below changes-
1) Validation logic changes
2) Function modification

### CODE REVIEW

File: src/main/java/com/stickynotes/createsticky/PostController.java [hunk 1]
```diff
@@ -9,22 +11,89 @@
...
+	/*
+	 * The method is responsible for controlling the creation of new user
+	 * @see com.stickynotes.services.CreateUserService #createUser(com.stickynotes.pojos.UserPojo)
+	 */
+	@RequestMapping(value="/addUser", method=RequestMethod.POST, headers="Accept=application/json")
+	@ResponseBody
+	public ResponseEntity<?> createUser(@RequestBody UserPojo userPojo){
+		
+		ExceptionResponse exceptionResponse= UserValidation.createUserValidation(userPojo);
+		if(exceptionResponse.getStatus()==false){
```
Comment (vavs0208) [inline src/main/java/com/stickynotes/createsticky/PostController.java]: Incorrect use of if loop.

File: src/main/java/com/stickynotes/dao/ChangeUserPasswordDao.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	 * Parameter com.stickynotes.pojos.UserPojo
+	 * Return    com.stickynotes.dto.UserPojo
+	 */
+	public UserDto changeUserPassword(UserPojo userPojo){
+		UserDto userDto=new UserDto();
+		ModelMapper modelMapper=new ModelMapper();
+		modelMapper.map(userPojo, userDto);
+		
+		try {
+			if(userPojo.getPassword().equals(userPojo.getNewPassword())){
```
Comment (vavs0208) [inline src/main/java/com/stickynotes/dao/ChangeUserPasswordDao.java]: Please ensure that this validation happens @ controller level;

File: src/main/java/com/stickynotes/dao/ChangeUserPasswordDao.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+					String password=userEntity.getPassword();
+					if(passwordEncoderMatcher.match(userPojo.getOldPassword(), password)){
+					
+						if(passwordEncoderMatcher.match(userPojo.getPassword(), password)){
+							userDto.setPresent(true);
+							userDto.setMessege(env.getProperty("samepassword"));
+							return userDto;
+						}
+						
+						userEntity.setPassword(passwordEncoderMatcher.hashPassword(userPojo.getPassword()));
```
Comment (vavs0208) [inline src/main/java/com/stickynotes/dao/ChangeUserPasswordDao.java]: Please use the update query rather than saving the entity

**Comment by Tirtha1388** [comment]
closing this and creating new PR

**Pull request closed without merging**